### PR TITLE
Firefly III

### PR DESCRIPTION
**Pull request by Barrow1990**
# Firefly III

# Description

"Firefly III" is a (self-hosted) manager for your personal finances. It can help you keep track of your expenses and income, so you can spend less and save more. Firefly III supports the use of budgets, categories and tags. Using a bunch of external tools, you can import data. It also has many neat financial reports available.

Firefly III should give you insight into and control over your finances. Money should be useful, not scary. You should be able to see where it is going, to feel your expenses and to... wow, I'm going overboard with this aren't I?

But you get the idea: this is your money. These are your expenses. Stop them from controlling you.

## Links to further information

- [X] Link to project homepage or github | [Homepage](https://www.firefly-iii.org) | [Github](https://github.com/firefly-iii/firefly-iii/)
- [X] Link to documentation | [Docs](https://docs.firefly-iii.org/?mtm_campaign=firefly-iii-org&mtm_kwd=top-link)
- [X] Link to installation instructions | [Installation Instructions](https://docs.firefly-iii.org/how-to/firefly-iii/installation/docker/)
- [X] Link to compose file | [Example Docker Compose File](https://raw.githubusercontent.com/firefly-iii/docker/main/docker-compose.yml)
- [X] App Request #381 & #382
- [ ] Documentation (To Be Completed)

# How Has This Been Tested?

## System Config
- Operating System
  - [X] Ubuntu 24.04

- Install Type
  - [X] Clean Install
  - [X] Re-Install
  - [X] With custom environment variables
     - Added SMTP Mail

# Other Information

- [X] Utilises MariaDB built in
- [X] Generates a unique MariaDB Password
- [X] Generates a unique AppKey password for the Firefly III Application
- [X] Installs the data importer #382 as default

### CODE REVIEW

**Comment by owine** [comment]
I don't think the .env files are needed as the ENVs have been reproduced as part of the defaults (which is our preferred method anyway.) Does the app require the presense of the env files?

**Comment by owine** [comment]
> > I don't think the .env files are needed as the ENVs have been reproduced as part of the defaults (which is our preferred method anyway.) Does the app require the presense of the env files?
> 
> @owine Apologies. Thought id only uploaded the relevant files. These have now been removed from the repo

Thanks - can you just make the cleanups suggested by ansible-lint:

https://github.com/saltyorg/Sandbox/pull/383/files#annotation_26204799530
https://github.com/saltyorg/Sandbox/pull/383/files#annotation_26204799534

One other change I will put up shortly as well and we should be good to merge.

**Comment by Barrow1990** [comment]
> > > I don't think the .env files are needed as the ENVs have been reproduced as part of the defaults (which is our preferred method anyway.) Does the app require the presense of the env files?
> > 
> > 
> > @owine Apologies. Thought id only uploaded the relevant files. These have now been removed from the repo
> 
> Thanks - can you just make the cleanups suggested by ansible-lint:
> 
> https://github.com/saltyorg/Sandbox/pull/383/files#annotation_26204799530 https://github.com/saltyorg/Sandbox/pull/383/files#annotation_26204799534
> 
> One other change I will put up shortly as well and we should be good to merge.

Changes have been made.

File: sandbox.yml [hunk 1]
```diff
@@ -57,6 +57,8 @@
     - { role: filebot, tags: ['filebot'] }
     - { role: filebrowser, tags: ['filebrowser'] }
     - { role: filezilla, tags: ['filezilla'] }
+    - { role: fireflyiii, tags: ['fireflyiii'] }
+    - { role: fireflyiii_importer, tags: ['fireflyiii_importer'] }
```
[the remark this answers is not in the document]
Comment (Barrow1990) [reply sandbox.yml:61]: In general you would think not but this package can be utilised from a separate system.
Also can just run that tag if they need to redeploy that app.
So leaning towards leave it in

**Review by Barrow1990 — COMMENTED** [review]
Reviewed changes. Leaving the firefly_importer in the available tags and updated the document to correctly match the required jinja2 syntax

**Pull request merged**